### PR TITLE
Support Content-Length and Last-Modified for webview local resources

### LINKED ISSUE #125992: Provide Last-Modified and Content-Length information for webview local resources
Since the only way to access local resources in a webview is via web requests proxied through `asWebviewUri`, it would be nice if the ServiceWorker handling these provided additional metadata in the headers, namely `Content-Length` and `Last-Modified`, whenever possible. With this feature, webview extensions can allocate buffers of the right size (length) and do some caching (mtime).

CC @mjbvz

### PR DESCRIPTION
**Pull request by Vtec234**
Closes #125992

To test (steps from @mjbvz):
1. At the top of your workspace, create markdown
2. Also create an image called `img.png`
3. In markdown preview, add the content:
```js
<script>
  fetch('img.png', {
    method: 'HEAD'
}).then(async (e) => {
    console.log(e.status);
    console.log(e.headers.get('content-type'));
    console.log(e.headers.get('content-length'));
    console.log(e.headers.get('last-modified'));
})
</script> 
```
4. Open the markdown preview and make sure scripts are enabled (there should be a warning if they are not)
5. Open the developer tools, check that the header contents have been printed and are correct
6. Same steps for a `GET` request

### CODE REVIEW

File: src/vs/workbench/contrib/webview/browser/pre/service-worker.js [hunk 1]
```diff
@@ -100,7 +100,9 @@ class RequestStore {
 
 /**
  * Map of requested paths to responses.
- * @typedef {{ type: 'response', body: any, mime: string, etag: string | undefined, } | { type: 'not-modified', mime: string } | undefined} ResourceResponse
+ * @typedef {{ type: 'response', body: Uint8Array, mime: string, etag: string | undefined, mtime: number | undefined } |
```
Comment (Vtec234) [inline src/vs/workbench/contrib/webview/browser/pre/service-worker.js:103]: Note the stronger type on `body: Uint8Array`. AFAICT that's the only type which is ever passed through in a response and rely on it being so to call `Uint8Array.byteLength`.

File: src/vs/workbench/contrib/webview/browser/pre/service-worker.js [hunk 2]
```diff
@@ -233,15 +235,19 @@ async function processResourceRequest(event, requestUrl) {
 			}
 		}
 
-		/** @type {Record<String, string>} */
+		/** @type {Record<string, string>} */
```
Comment (Vtec234) [inline src/vs/workbench/contrib/webview/browser/pre/service-worker.js:238]: typo?

File: src/vs/workbench/contrib/webview/browser/resourceLoading.ts [hunk 3]
```diff
@@ -50,7 +52,7 @@ export async function loadLocalResource(
 	logService: ILogService,
 	token: CancellationToken,
 ): Promise<WebviewResourceResponse.StreamResponse> {
-	logService.debug(`loadLocalResource - being. requestUri=${requestUri}`);
+	logService.debug(`loadLocalResource - begin. requestUri=${requestUri}`);
```
Comment (Vtec234) [inline src/vs/workbench/contrib/webview/browser/resourceLoading.ts:55]: typo?

**Comment by mjbvz** [comment]
Thanks! Will be in the next VS Code insiders build and is scheduled for 1.58

**Comment by Vtec234** [comment]
Thanks! Is the macOS test failure not a concern?

**Pull request merged**